src/state/nonces/index.ts

```diff
@@ -20,11 +22,36 @@ export async function getNextNonce({ address, chainId }: { address: string; chai
   const { getNonce } = nonceStore.getState();
   const localNonceData = getNonce({ address, chainId });
   const localNonce = localNonceData?.currentNonce || 0;
-  const provider = getProvider({ chainId });
-  const txCountIncludingPending = await provider.getTransactionCount(address, 'pending');
-  if (!localNonce && !txCountIncludingPending) return 0;
-  const ret = Math.max(localNonce + 1, txCountIncludingPending);
-  return ret;
+  const provider = getBatchedProvider({ chainId });
+  const privateMempoolTimeout = chainsPrivateMempoolTimeout[chainId];
+
+  const pendingTxCountRequest = provider.getTransactionCount(address, 'pending');
+  const latestTxCountRequest = provider.getTransactionCount(address, 'latest');
+  const [pendingTxCountFromPublicRpc, latestTxCountFromPublicRpc] = await Promise.all([pendingTxCountRequest, latestTxCountRequest]);
```

Comment:
batched provider will batch requests sent on the provider in the same event loop